src/equations/compressible_euler_multicomponent_2d.jl

```diff
@@ -1,67 +1,96 @@
 
 !!! warning "Experimental code"
     This system of equations is experimental and can change any time.
 
-Multicomponent version of the compressible Euler equations for an ideal gas in two space dimensions.
+Multicomponent version of the compressible Euler equations 
+```math
+\begin{equation*}
+\partial t
+\begin{pmatrix}
+\rho v_1 \\ \rho v_2 \\ E \\ \rho_1 \\ \rho_2 \\ \vdots \\ \rho_{n}
+\end{pmatrix}
++
+\partial x
+\begin{pmatrix}
+\rho v_1^2 + p \\ \rho v_1 v_2 \\ (E+p) v_1 \\ \rho_1 v_1 \\ \rho_2 v_1 \\ \vdots \\ \rho_{n} v_1
+\end{pmatrix}
++
+\partial y
+\begin{pmatrix}
+\rho v_1 v_2 \\ \rho v_2^2 + p \\ (E+p) v_2 \\ \rho_1 v_2 \\ \rho_2 v_2 \\ \vdots \\ \rho_{n} v_2
+\end{pmatrix}
+= 
+\begin{pmatrix}
+0 \\ 0 \\ 0 \\ 0 \\ 0 \\ \vdots \\ 0
+\end{pmatrix}
+\end{equation*}
+```
+for calorically perfect gas in two space dimensions.
+
+The specific heat ratios 'gammas' and the gas constants 'gas_constants' in [kJ/(kg*K)]
+should be passed as tuples e.g. gammas=(1.4, 1.667) or gas_constants=(0.4,) in the one component case.
+
+The remaining variables like the specific heats at constant volume 'cv' or the specific heats at 
+constant pressure 'cp' are then calculated considering a calorically perfect gas.
 """
-struct CompressibleEulerMulticomponentEquations2D{RealT<:Real} <: AbstractCompressibleEulerMulticomponentEquations{2, 5}
-  gamma1          ::RealT 
-  gamma2          ::RealT
-  gas_constant1   ::RealT
-  gas_constant2   ::RealT
-  cv1             ::RealT
-  cv2             ::RealT
-  cp1             ::RealT
-  cp2             ::RealT
-end
 
-function CompressibleEulerMulticomponentEquations2D()
+struct CompressibleEulerMulticomponentEquations2D{NCOMP, NVARS, RealT<:Real} <: AbstractCompressibleEulerMulticomponentEquations{2, NVARS}
+  gammas            ::SVector{NCOMP, RealT} 
+  gas_constants     ::SVector{NCOMP, RealT}
+  cv                ::SVector{NCOMP, RealT}
+  cp                ::SVector{NCOMP, RealT}
 
-  # Set ratio of specific heat of the gas per species 
-  # (It holds: gamma = cp/cv) (Can be used for consistency check!)
-  gamma1  = 1.4
-  gamma2  = 1.4
+  function CompressibleEulerMulticomponentEquations2D{NCOMP, NVARS, RealT}(gammas, gas_constants) where {NCOMP, NVARS, RealT<:Real} 
   
-  # Set specific gas constant with respect to the molar mass per species 
-  # (It holds: gas_constant = R/M, with R = molar gas constant, M = molar mass) 
-  # (For a calorically and thermally perfect gas Mayer's relation holds: gas_constant = cp - cv) (Can be used for consistency check!)
-  gas_constant1     = 0.4
-  gas_constant2     = 0.4
+    new(gammas, # input
+        gas_constants, # input
+        gas_constants./(gammas.-1), # compute 'cv'
+        gas_constants .+ gas_constants./(gammas .-1) # compute 'cp'
+       )
+  end
+
+end
+
 
-  # Set specific heat for a constant volume per species
-  cv1    = 1.0
-  cv2    = 1.0
+function CompressibleEulerMulticomponentEquations2D(; gammas, gas_constants)
```

Comment:
Sorry for not realizing this earlier, but all the variable transformations and code checking you are doing in this outer ctor should really be moved into the inner ctor. Right now, nothing stops a user from calling the inner ctor directly and providing bogus/mismatching type information. Since enforcing invariants is one of the two main raisons d'être, I suggest to move all this code checking there (see also https://docs.julialang.org/en/v1/manual/constructors/#man-inner-constructor-methods).

The only good reason for keeping the inner ctor as it is would be if there are some use cases where it might make sense to violate the constraints imposed here, but then I suggest to document it somewhere such that users know that they really should use *this* ctor here.

Reply:
done in 4db20a5